homeassistant/components/tplink/__init__.py

```diff
@@ -102,7 +103,20 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             f"Unexpected device found at {host}; expected {entry.unique_id}, found {found_mac}"
         )
 
-    hass.data[DOMAIN][entry.entry_id] = TPLinkDataUpdateCoordinator(hass, device)
+    parent_coordinator = TPLinkDataUpdateCoordinator(hass, device, timedelta(seconds=5))
+    child_coordinators: list[TPLinkDataUpdateCoordinator] = []
+
+    if device.is_strip:
+        child_coordinators = [
+            # The child coordinators only update energy data so we can
+            # set a longer update interval to avoid flooding the device
+            TPLinkDataUpdateCoordinator(hass, child, timedelta(minutes=2))
```

Comment:
I'm wondering if this is too long update period, assuming someone wants to use the power usage for automations? Also, wouldn't this also affect the child state updates even when they do not perform any I/O?

Alternatively, we could optimize the "state" updates by leveraging the data returned in the sysinfo response to and drop this to be 10-15s as it used to be.

Reply:
It's only the power since the child on/off state comes from the parent update which is still every 5s

Reply:
I changed it to 1m.  Its still much more stable but it still does hammer the state machine a bit